src/environment.cpp

```diff
@@ -1256,6 +1256,50 @@ void ServerEnvironment::step(float dtime)
 		*/
 		removeRemovedObjects();
 	}
+
+	/*
+		Manage particle spawner expiration
+	*/
+	if (m_particle_management_interval.step(dtime, 1.0)) {
```

Comment:
Hmm, yes. Isn't that the case with the original code as well in that case (line 1275)?